cluster/client.go

```diff
@@ -145,6 +145,35 @@ func (c *Client) GetCommitIndex(nodeAddr string, retries int, timeout time.Durat
 	return a.CommitIndex, nil
 }
 
+func (c *Client) GetLeaderCommitIndex(nodeAddr string, retries int, trustLeaderLease bool, timeout time.Duration) (uint64, error) {
```

Comment:
Instead of adding a new function, just modify the existing `GetCommitIndex` so it takes a new boolean called `verifyLeader`. If set, do your leader check.

I am not concerned about this software talking to nodes that don't understand the new field in the proto because:

- I tell folks not to run clusters with a mix of versions, except during the small window they are upgrading
- I tell folks that if they do run clusters with a mix of versions they might get a query error until all nodes are upgraded.

This will be the case here, so I'm not worried about changing existing RPC calls.

Reply:
See https://github.com/rqlite/rqlite/blob/master/cluster/client.go#L129 to see what should change, and not add a new RPC call.

Reply:
As a result, you'll end up with something like this:
```
message NodeMeta {
    string url = 1;
    uint64 commit_index = 2;
    bool verify_leader = 3;
}
```

As I said, I'm not super-concerned about changes like this, as it's purely implementation details, and not issue in practise. Folks just need to upgrade all nodes, and not run mixed versions. I always make this clear in release notes.